src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,46 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserValidationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN = 6;
+    private static final int MIN_PASSWORD = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserValidationException("User can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new UserValidationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN) {
+            throw new UserValidationException("Login is too short!");
+        }
+        if (!Storage.people.isEmpty()) {
+            if (storageDao.get(user.getLogin()) != null) {
+                throw new UserValidationException("User with login: \""
+                        + user.getLogin() + "\" already exists");
+            }
+        }
```

Comment:
The check `if (!Storage.people.isEmpty())` is redundant because `storageDao.get(user.getLogin())` will return `null` if the user does not exist, regardless of whether the storage is empty. Consider removing this check to simplify the code.